instructor/common.py

```diff
@@ -370,10 +370,29 @@ def assignments_edit_treatments(request):
         input_headers.extend(['Duration', 'Duration Units'])
     else:
         input_headers.extend(['Start Time', 'Duration', 'Time Units'])
-
-    DrugFormSet = modelformset_factory(models.Drug, extra=1, can_delete=True, exclude=drug_formset_exclude)
-    TemperatureFormSet = modelformset_factory(models.Temperature, extra=1, can_delete=True, exclude=['assignment'])
-    CollectionTimeFormSet = modelformset_factory(models.CollectionTime, extra=1, can_delete=True, exclude=['assignment'])
+    # If instructor clicked ADD, add extra form
+    drug_extra_form = 0
+    temperature_extra_form = 0
+    collection_extra_form = 0
+    if 'add_drug' in request.POST or not models.Drug.objects.filter(assignment=assignment):
```

Comment:
It's more efficient to add `.exists()` at the end of the QuerySet because it will check for existence in SQL rather than in Python. The way it is right now it will get the rows from the database and then apply `not` to it to convert to a boolean, with zero rows being a False and more than zero being true.

If you're ever curious about the SQL being generated you can turn on SQL logging in `settings.py`, run `./manage.py shell` and try out different QuerySets to see what they produce. To turn on SQL logging I made this adjustment to `LOGGING` in `settings.py`:

    LOGGING = {
        'version': 1,
        'disable_existing_loggers': False,
        'filters': {
            'require_debug_false': {
                '()': 'django.utils.log.RequireDebugFalse'
            }
        },
        'handlers': {
            'mail_admins': {
                'level': 'ERROR',
                'filters': ['require_debug_false'],
                'class': 'django.utils.log.AdminEmailHandler'
            },
            'console': {
                'level': 'DEBUG',
                'class': 'logging.StreamHandler',
                'formatter': 'verbose'
            }
        },
        'formatters': {
            'verbose': {
                'format': (
                    '[%(asctime)s] %(levelname)s %(process)d [%(name)s] '
                    '%(filename)s:%(lineno)d - '
                    '[{hostname}] - %(message)s'
                ).format(hostname="VAGRANT"),
                'datefmt': '%Y-%m-%d %H:%M:%S'
            }
        },
        'loggers': {
            '': {
                'handlers': ['console'],
                'level': 'DEBUG'
            },
            'django.request': {
                'handlers': ['mail_admins'],
                'level': 'ERROR',
                'propagate': True,
            },
        }
    }

This logs absolutely everything to console which may be a bit much but can be helpful in development.